File: src/main/java/com/notehub/notehub/user/UserService.java
package com.notehub.notehub.user;

import java.util.Optional;
import java.util.UUID;

import org.springframework.data.domain.Page;

public interface UserService {

    Optional<User> findById(UUID id);

    Optional<User> findByUsername(String username);

    Optional<User> findByEmail(String email);

    Page<User> listUsers(int page, int size, String direction, String sortBy);

    User createUser(User user);

    User updateUser(UUID id, User updatedUser);

    void deleteUser(UUID id);
}

File: frontend/src/features/user/services/UserService.ts
import { api as API } from "../../../shared/util/api";
import { UserDTO } from "../types";

export const UserService = {
  async fetchUser() {
    const response = await API.get<UserDTO>("/user");
    return response.data;
  },

  async updateUser(userDTO: UserDTO) {
    const response = await API.patch("/user", userDTO);
    return response.data;
  },

  async deleteUser() {
    const response = await API.delete("/user");
    return response.data;
  },
};
